recipes/boost/all/conanfile.py

```diff
@@ -59,42 +87,67 @@ class BoostConan(ConanFile):
+    for _name in CONFIGURE_OPTIONS:
+        options["without_{}".format(_name)] = [True, False]
 
     default_options = {
-        'shared': False,
-        'header_only': False,
-        'error_code_header_only': False,
-        'system_no_deprecated': False,
-        'asio_no_deprecated': False,
-        'filesystem_no_deprecated': False,
-        'fPIC': True,
-        'layout': 'system',
-        'magic_autolink': False,
-        'python_executable': 'None',
-        'python_version': 'None',
-        'namespace': 'boost',
-        'namespace_alias': False,
-        'multithreading': True,
-        'zlib': True,
-        'bzip2': True,
-        'lzma': False,
-        'zstd': False,
-        'segmented_stacks': False,
+        "shared": False,
+        "header_only": False,
+        "error_code_header_only": False,
+        "system_no_deprecated": False,
+        "asio_no_deprecated": False,
+        "filesystem_no_deprecated": False,
+        "fPIC": True,
+        "layout": "system",
+        "magic_autolink": False,
+        "python_executable": "None",
+        "python_version": "None",
+        "namespace": "boost",
+        "namespace_alias": False,
+        "multithreading": True,
+        "zlib": True,
+        "bzip2": True,
+        "lzma": False,
+        "zstd": False,
+        "segmented_stacks": False,
         "debug_level": 0,
-        'pch': True,
-        'extra_b2_flags': 'None',
-        "i18n_backend": 'iconv',
+        "pch": True,
+        "extra_b2_flags": "None",
+        "i18n_backend": "iconv",
     }
+    for _name in CONFIGURE_OPTIONS:
+        default_options["without_{}".format(_name)] = False
+    for _name in ("graph_parallel", "mpi", "python",):
+        default_options["without_{}".format(_name)] = True
 
-    for libname in lib_list:
-        if libname != "python":
-            default_options.update({"without_%s" % libname: False})
-    default_options.update({"without_python": True})
     short_paths = True
     no_copy_source = True
     exports_sources = ['patches/*']
 
+    def export(self):
+        self.copy(self._dependency_filename, src="dependencies", dst="dependencies")
+
+    @property
+    def _dependency_filename(self):
+        return "dependencies-{}.yml".format(self.version)
+
+    @property
+    def _dependencies(self):
+        dependencies_filepath = os.path.join(self.recipe_folder, "dependencies", self._dependency_filename)
+        if not os.path.isfile(dependencies_filepath):
+            raise ConanException("Cannot find {}".format(dependencies_filepath))
+        return yaml.load(open(dependencies_filepath))
+
+    def _iter_modules(self):
+        tree = {k: v[:] for k, v in self._dependencies["dependencies"].items()}
+        while tree:
+            nodeps = set(k for k, v in tree.items() if not v)
+            if not nodeps:
+                raise ConanException("cyclic dependency tree detected")
+            for nodep in nodeps:
+                yield nodep
```

Comment:
What does this do? 🤓

Reply:
It's just a python way to return an iterator.
It can be rewritten as returning a list/tuple, but meh.